persistence/src/main/scala/akka/persistence/kafka/journal/KafkaJournal.scala

```diff
@@ -1,240 +1,32 @@
 package akka.persistence.kafka.journal
 
-import akka.actor.ActorSystem
-import akka.persistence.journal.AsyncWriteJournal
-import akka.persistence.{AtomicWrite, PersistentRepr}
-import cats.Parallel
-import cats.effect._
-import cats.implicits._
-import com.evolutiongaming.catshelper.{FromFuture, Log, LogOf, ToFuture}
-import com.evolutiongaming.kafka.journal._
-import com.evolutiongaming.kafka.journal.util.CatsHelper._
-import com.evolutiongaming.kafka.journal.util.PureConfigHelper._
-import com.evolutiongaming.scassandra.CassandraClusterOf
-import com.evolutiongaming.smetrics.MeasureDuration
+import cats.effect.{IO, Resource}
+import com.evolutiongaming.catshelper.ToTry
+import com.evolutiongaming.kafka.journal.{JournalReadWrite, JsonCodec, Payload}
+import com.evolutiongaming.kafka.journal.Events._
+import com.evolutiongaming.kafka.journal.PayloadAndType.PayloadJson._
 import com.typesafe.config.Config
-import pureconfig.ConfigSource
 
-import scala.collection.immutable.Seq
-import scala.concurrent.duration._
-import scala.concurrent.{Await, ExecutionContextExecutor, Future}
 import scala.util.Try
 
-class KafkaJournal(config: Config) extends AsyncWriteJournal { actor =>
+class KafkaJournal(config: Config) extends BaseKafkaJournal[Payload](config) {
```

Comment:
could you please explain me what kind of naming strategy you have in mind for this?

Reply:
No specific reason for such names. Please suggest what you'd like the names to be